src/realm/object-store/shared_realm.cpp

```diff
@@ -1278,11 +1296,26 @@ void Realm::close()
     do_invalidate();
 
     m_binding_context = nullptr;
+
+    m_coordinator->close();
```

Comment:
This is the most important change. It seems we have a cyclic dependency, `RealmCoordinator` needs `Realm` and `Realm` needs `RealmCoordinator`. We never release the realm resources when we delete coordinator. 
This: 
`m_coordinator = nullptr` does everything but closing the file. It seems this is done on purpose, so I just added ` m_coordinator->close()' inside the `Realm::close`, which is called explicitly.